src/WebJobs.Script/StorageProvider/TableServiceClientProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Azure.Core;
+using Azure.Data.Tables;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class TableServiceClientProvider : StorageClientProvider<TableServiceClient, TableClientOptions>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableServiceClientProvider"/> class that uses the registered Azure services to create a <see cref="TableServiceClient"/>.
+        /// </summary>
+        /// <param name="componentFactory">The Azure factory responsible for creating clients. <see cref="AzureComponentFactory"/>.</param>
+        /// <param name="logForwarder">Log forwarder that forwards events to ILogger. <see cref="AzureEventSourceLogForwarder"/>.</param>
+        public TableServiceClientProvider(AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder) : base(componentFactory, logForwarder)
+        {
+        }
+
+        protected override TableServiceClient CreateClient(IConfiguration configuration, TokenCredential tokenCredential, TableClientOptions options)
+        {
+            // If connection string is present, it will be honored first; bypass creating a serviceUri
+            if (!IsConnectionStringPresent(configuration))
+            {
+                var serviceUri = configuration.Get<StorageServiceUriOptions>().GetTableServiceUri();
```

Comment:
Is it possible that `configuration.Get<StorageServiceUriOptions>()` could return `null`? Should we do a null check before calling the `GetTableServiceUri` method?